tutorials/mission_specific_analyses/rxte/visualize-rxte-spectra.md

```diff
@@ -0,0 +1,379 @@
+
+<span style="color:red">This solution is currently too slow to include in the final version of this 
+demonstration, which I hope is due to my ignorance of PyVO</span>
+
+
+```{code-cell}
+rel_dls = {}
+
+with tqdm(desc="Identifying XTE PCA spectra S3 URIs", total=len(results)) as uri_prog:
+    for res in results:
+        rel_dls[res['obsid']] = {}
+        for res_dls in res.getdatalink().iter_datalinks():
+            for all_prod_dls in res_dls.iter_datalinks():
+                for prod_dl in all_prod_dls.bysemantics(search_sem):
+                    # This use of ast converts the string representation of a Python dictionary stored under
+                    #  the 'cloud_access' key to an actual, addressable, Python dictionary
+                    cloud_dict = ast.literal_eval(prod_dl['cloud_access'])
+                    rel_uri = cloud_dict['aws']['key']
+                    rel_dls[res['obsid']][conv_desc[prod_dl['description']]] = "s3://nasa-heasarc/" + rel_uri
+
+        uri_prog.update(1)
+        
+# Show a single example of the datalinks
+rel_dls[results[0]['obsid']]
+```
+
+## 2. Acquiring the data
+We now know where the relevant RXTE-PCA spectra are stored in the HEASARC S3 bucket, and will proceed to download 
+them for local use. 
+
+***Many workflows are being adapted to stream remote data directly into memory*** (RAM), rather than
+downloading it onto disk storage, *then* reading into memory - PyXspec does not yet support this way of 
+operating, but our demonstrations will be updated when it does.
+
+### Creating a storage directory
+
+Our first task is to set up a directory in which to store the download source spectra, background spectra, and 
+response files.
+
+```{code-cell}
+stor_dir = "xte_data/pca_spec/"
+os.makedirs(stor_dir, exist_ok=True)
+```
+
+### Setting up an S3 file system
+
+Now we make use of a Python module called `s3fs`, which allows us to interact with files stored on Amazon's S3 
+platform through Python commands. Our use of this module for this demonstration will be very simple, as we're only
+going to download files.
+
+We create an `S3FileSystem` object, which will accept the datalinks constructed in the last step as a form of file 
+path and allow us to download the files to local storage - **note the `anon=True` argument, as access to the HEASARC
+S3 bucket will fail without it**.
+
+```{code-cell}
+s3 = S3FileSystem(anon=True)
+```
+
+
+### Downloading our spectra
+
+```{code-cell}
+local_prod_paths = {}
+
+for oi, cur_rel_dl in rel_dls.items():    
+    cur_stor_dir = os.path.join(stor_dir, oi)
+    os.makedirs(cur_stor_dir, exist_ok=True)
+
+    local_prod_paths[oi] = {cur_ft: os.path.join(cur_stor_dir, cur_uri.split('/')[-1]) for cur_ft, cur_uri in cur_rel_dl.items()}
+    file_exists = [os.path.exists(f) for f in local_prod_paths[oi].values()]
+    
+    if not all(file_exists):
+        s3.get(list(cur_rel_dl.values()), cur_stor_dir)
```

Comment:
`Heasarc.download_data(.., host='aws')` in astroquery would do this, which hides the details from the user